### PR TITLE
Implement screen delete/add

### PR DESCRIPTION
**Pull request by sergey-borovkov**
## What does this PR do?
Adds screen delete/add commands.
## How has this been tested?
Added unit tests to make sure correct requests are sent.
https://ph.wireload.net/T6411
## Checklist before merging

- [x] If have added tests.
- [x] Is this a new feature? If yes, please write one phrase about this update.
- [ ] I've attached relevant screenshots (if relevant).

### CODE REVIEW

**Comment by sergey-borovkov** [comment]
> We probably need some kind of manual verification before you delete a screen, like entering the screen's name or similar.

ok, let me add it later in a separate PR.

**Comment by vpetersson** [comment]
> > We probably need some kind of manual verification before you delete a screen, like entering the screen's name or similar.
> 
> ok, let me add it later in a separate PR.

I'd prefer if we fixed it before. I know the code is still in alpha but still important.

**Comment by sergey-borovkov** [comment]
Ok, sure. Will fix it soon.

**Review by vpetersson — APPROVED** [review]
LGTM with regards to messaging. Can't comment on code.

**Comment by renatgalimov** [comment]
@sergey-borovkov , please add unit-tests for this feature

>You are about to delete the screen named "sergey pi4 dev 1".  This operation cannot be reversed.
Enter the screen name to confirm the screen deletion

File: src/commands.rs [hunk 1]
```diff
@@ -127,6 +131,46 @@ impl ScreenCommand {
...
     }
+
+    pub fn add(
+        &self,
+        pin: &str,
+        maybe_name: Option<String>,
+    ) -> anyhow::Result<Screens, CommandError> {
+        let url = self.authentication.config.url.clone() + "/v3/screens/";
+        let mut payload = HashMap::new();
+        payload.insert("pin".to_string(), pin.to_string());
```
Comment (korvyashka) [inline src/commands.rs:142]: If pin is anyway converted from &str to String - may be function should accept String instead.

File: src/commands.rs [hunk 2]
```diff
@@ -127,6 +131,46 @@ impl ScreenCommand {
...
+            payload.insert("name".to_string(), name);
+        }
+        let response = self
+            .authentication
+            .build_client()?
+            .post(url)
+            .json(&payload)
+            .send()?;
+        if response.status().as_u16() != 201 {
+            return Err(CommandError::WrongResponseStatus(
```
Comment (korvyashka) [inline src/commands.rs:153]: Do we want to parse errors sent here?

File: src/commands.rs [hunk 3]
```diff
@@ -127,6 +131,46 @@ impl ScreenCommand {
...
+        }
+
+        // Our newer endpoints all return arrays so let's just convert the output from v3 to be the same
+        let mut array: Vec<serde_json::Value> = Vec::new();
+        array.insert(0, serde_json::from_str(&response.text()?)?);
+        Ok(Screens::new(serde_json::Value::Array(array)))
+    }
+
+    pub fn delete(&self, id: &str) -> anyhow::Result<(), CommandError> {
+        let url = self.authentication.config.url.clone() + "/v3/screens/" + id + "/";
```
Comment (korvyashka) [inline src/commands.rs:165]: format! can be used.

File: src/commands.rs [hunk 4]
```diff
@@ -169,6 +214,35 @@ mod tests {
...
....399+00:00\",\"uptime\":\"230728.38\",\"load_avg\":\"0.14\",\"signal_strength\":null,\"interface\":\"eth0\",\"debug\":false,\"location\":\"Kamsko-Ust'inskiy rayon, Russia\",\"team\":\"016343c2-82b8-0000-a121-e30f1035875e\",\"timezone\":\"Europe/Moscow\",\"type\":\"hardware\",\"hostname\":\"srly-4shnfrdc5cd2p0p\",\"ws_open\":false,\"status\":\"Offline\",\"last_screenshot\":\"https://us-assets.screenlyapp.com/01CD1W50NR000A28F31W83B1TY/screenshots/01F98G8MJB6FC809MGGYTSWZNN/5267668e6db35498e61b83d4c702dbe8\",\"in_sync\":false,\"software_version\":\"Screenly 2 Player\",\"hardware_version\":\"Raspberry Pi 3B\",\"config\":{\"hdmi_mode\": 34, \"hdmi_boost\": 2, \"hdmi_drive\": 0, \"hdmi_group\": 0, \"verify_ssl\": true, \"audio_output\": \"hdmi\", \"hdmi_timings\": \"\", \"overscan_top\": 0, \"overscan_left\": 0, \"use_composite\": false, \"display_rotate\": 0, \"overscan_right\": 0, \"overscan_scale\": 0, \"overscan_bottom\": 0, \"disable_overscan\": 0, \"shuffle_playlist\": false, \"framebuffer_width\": 0, \"use_composite_pal\": false, \"framebuffer_height\": 0, \"hdmi_force_hotplug\": true, \"use_composite_ntsc\": false, \"hdmi_pixel_encoding\": 0, \"play_history_enabled\": false}}");
```
Comment (korvyashka) [inline src/commands.rs:233]: Maybe easier to build HashMap and convert to json - to improve readability.

Comment (sergey-borovkov) [reply src/commands.rs:233]: It is in the tests does not really matter. I will change when I touch this next maybe

File: src/main.rs [hunk 5]
```diff
@@ -79,32 +139,48 @@ fn main() {
...
+            ScreenCommands::Add { pin, name, json } => {
+                let screen_command = commands::ScreenCommand::new(authentication);
+                handle_command_execution_result(screen_command.add(pin, name.clone()), json);
+            }
+            ScreenCommands::Delete { uuid } => {
+                let screen_command = commands::ScreenCommand::new(authentication);
+                match get_screen_name(uuid, &screen_command) {
+                    Ok(name) => {
+                        println!("You are about to delete the screen named \"{}\".  This operation cannot be reversed.", name);
+                        print!("Enter the screen name to confirm the screen deletion: ");
```
[the remark this answers is not in the document]
Comment (sergey-borovkov) [reply src/main.rs:157]: Not for now.

File: src/commands.rs (hunk 2 above)
Comment (sergey-borovkov) [reply src/commands.rs:153]: No

File: src/commands.rs (hunk 1 above)
Comment (sergey-borovkov) [reply src/commands.rs:142]: I think it's a best practice to use &str but I am not 100 percent sure

**Review by renatgalimov — APPROVED** [review]
Created #11

**Pull request merged**